openpype/hooks/pre_add_last_workfile_arg.py

```diff
@@ -44,10 +44,15 @@ def execute(self):
 
         # Determine whether to open workfile post initialization.
         if self.host_name == "maya":
-            key = "open_workfile_post_initialization"
-            if self.data["project_settings"]["maya"][key]:
+            keys = [
+                "open_workfile_post_initialization", "explicit_plugins_loading"
+            ]
```

Comment:
Still feels weird that `explict_plugins_loading` needs to be checked here but I can understand why it's needed.

Reply:
Yup agreed. The alternative would be have duplicated code in a `pre_explicit_plugins_loading.py` hook?

Reply:
Sort of, yes. I was thinking doing this:

```python
from openpype.lib import PreLaunchHook


class MayaLaunchWorkfilePostInitialization(PreLaunchHook):
    """Define whether open last workfile should run post initialize"""

    # Before AddLastWorkfileToLaunchArgs
    order = 9
    app_groups = ["maya"]
    
    def execute(self):
    
        # Ignore if there's no last workfile to start
        if not self.data.get("start_last_workfile"):
            return
    
        
        maya_settings = self.data["project_settings"]["maya"]
        enabled = maya_settings["open_workfile_post_initialization"]
        if enabled:
            # Force disable the `AddLastWorkfileToLaunchArgs`
            self.data.pop("start_last_workfile")
            
            self.launch_context.env["OPENPYPE_OPEN_WORKFILE_POST_INITIALIZATION"] = "1"
            
            
class MayaPreAutoLoadPlugins(PreLaunchHook):
    """Define -noAutoloadPlugins command flag."""

    # Before AddLastWorkfileToLaunchArgs
    order = 9
    app_groups = ["maya"]
    
    def execute(self):
    
        # Ignore if there's no last workfile to start
        if not self.data.get("start_last_workfile"):
            return
    
        
        maya_settings = self.data["project_settings"]["maya"]
        enabled = maya_settings["explicit_plugins_loading"]["enabled"]
        if enabled:
            # Force disable the `AddLastWorkfileToLaunchArgs`
            self.data.pop("start_last_workfile")
            self.launch_context.env["OPENPYPE_OPEN_WORKFILE_POST_INITIALIZATION"] = "1"
```

And if you don't want _that_ code duplication then at least you could merge these two together to still keep the logic close to the Maya integration and not mixed into the `AddLastWorkfileToLaunchArgs` pre launch hook.

Reply:
Done

Reply:
Did my pseudocode work out of the box? Haha, that's crazy! 😄

Reply:
Lol, yeah.